src/kw_manager.sh

```diff
@@ -265,15 +282,245 @@ function rename_kw_manager_group()
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function split_contact_infos()
+{
+  local contacts_list="$1"
+  local -n contacts_array="$2"
+  local contacts_infos_array
+  local contact_info
+  local ret
+  local name
+  local email
+  local added_contacts=0
+
+  readarray -t contacts_infos_array < <(awk -v RS=", " '{print}' <<< "$contacts_list")
+
+  for contact_info in "${contacts_infos_array[@]}"; do
+    if [[ -z "$contact_info" ]]; then
+      continue
+    fi
+
+    check_infos_sintaxe "$contact_info" || return 22 #EINVAL
+
+    email="$(cut --delimiter='<' --fields=2 <<< "$contact_info" | cut --delimiter='>' --fields=1)"
+    name="$(cut --delimiter='<' --fields=1 <<< "$contact_info")"
+
+    validate_contact_infos "$email" "$name" || return 22 #EINVAL
+
+    contacts_array["$email"]="$name"
+    ((added_contacts++))
+  done
+
+  if [[ "$added_contacts" -ne "${#contacts_array[@]}" ]]; then
+    complain 'Some of the contacts must have a repeated email'
+    return 22 #EINVAL
+  fi
+
+  return 0
+}
+
+# This function validate if the string with the contact infos is valid
+#
+# @contact_info: The string with the contact info,
+# formed as: "CONTACT_NAME <CONTACT_EMAIL>"
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function check_infos_sintaxe()
+{
+  local contact_info="$1"
+  local lt_pos
+  local lt_count
+  local gt_pos
+  local gt_count
+
+  gt_count=$(str_count_char_repetition "$contact_info" '>')
+  lt_count=$(str_count_char_repetition "$contact_info" '<')
+
+  gt_pos=$(str_get_char_position "$contact_info" '>')
+  lt_pos=$(str_get_char_position "$contact_info" '<')
+
+  if [[ "$lt_count" -eq 0 ]]; then
+    complain 'Syntax error in the contacts list, there is a missing "<" in some of the contacts <email>'
+    return 22 #EINVAL
+  elif [[ "$gt_count" -eq 0 ]]; then
+    complain 'Syntax error in the contacts list, there is a missing ">" in some of the contacts <email>'
+    return 22 #EINVAL
+  elif [[ "$lt_pos" -gt "$gt_pos" ]]; then
+    complain 'Syntax error in the contacts list, the contact info should be like: name <email>'
+    return 22 #EINVAL
+  elif [[ "$lt_count" -ne 1 ]]; then
+    complain 'Syntax error in the contacts list, there is a remaining "<" in some of the contacts <email>'
+    return 22 #EINVAL
+  elif [[ "$gt_count" -ne 1 ]]; then
+    complain 'Syntax error in the contacts list, there is a remaining ">" in some of the contacts <email>'
+    return 22 #EINVAL
+  fi
+
+  return 0
+}
+
+# This function check if the contacts infos email and name are valid
+#
+# @email: The contact email
+# @name: The contact name
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function validate_contact_infos()
+{
+  local email="$1"
+  local name="$2"
+
+  if [[ -z "$email" || -z "$name" ]]; then
+    complain 'Some of the contact names or emails must be empty'
+    return 22 #EINVAL
+  fi
+
+  validate_email "$email" || return 22 #EINVAL
```

Comment:
this function should be generic, maybe you can make a PR that makes this part of the kw_str.